packages/ember-data/lib/system/store.js

```diff
@@ -1445,8 +1447,9 @@ Store = Service.extend({
   },
 
   modelFactoryFor: function(key) {
-    if (this.container.has('model:' + key)) {
-      return this.container.lookupFactory('model:' + key);
+    var singularized = singularize(key);
```

Comment:
In some places we're using singularized, and other places fully normalizing with this._normalizeTypeKey.  Shouldn't we normalize the same way throughout?